main.go

```diff
@@ -232,6 +232,15 @@ func (h k8sHelpers) GetNode(cli *client.Client) (*api.Node, error) {
 }
 
 func (h k8sHelpers) AddLabels(n *api.Node, labels Labels) {
+	// Remove nfd node labels that no longer apply
+	for k := range n.Labels {
+		// If the label is not in the labels we are adding
+		// and it is an nfd label, delete it.
```

Comment:
Nit: Consider changing comment to `If the label was not discovered in the current run of node-feature-discovery and if it has the prefix, delete it from the list of node labels` or something similar.